orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/CheckPreconditionsStage.groovy

```diff
@@ -16,42 +16,38 @@
 
 package com.netflix.spinnaker.orca.pipeline
 
-import com.netflix.spinnaker.orca.DefaultTaskResult
 import com.netflix.spinnaker.orca.Task
-import com.netflix.spinnaker.orca.TaskResult
+import com.netflix.spinnaker.orca.pipeline.model.Execution
 import com.netflix.spinnaker.orca.pipeline.model.Stage
 import com.netflix.spinnaker.orca.pipeline.tasks.PreconditionTask
-import org.springframework.batch.core.Step
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.stereotype.Component
 
 @Component
-class CheckPreconditionsStage extends ParallelStage implements StepProvider {
+class CheckPreconditionsStage implements BranchingStageDefinitionBuilder {
+
   static final String PIPELINE_CONFIG_TYPE = "checkPreconditions"
 
-  @Autowired
-  List<PreconditionTask> preconditionTasks
+  private final List<? extends PreconditionTask> preconditionTasks
 
-  CheckPreconditionsStage() {
-    super(PIPELINE_CONFIG_TYPE)
+  @Autowired
+  CheckPreconditionsStage(List<? extends PreconditionTask> preconditionTasks) {
+    this.preconditionTasks = preconditionTasks
   }
 
   @Override
-  List<Step> buildSteps(Stage stage) {
+  def <T extends Execution<T>> void taskGraph(Stage<T> stage, TaskNode.Builder builder) {
```

Comment:
Groovy struggles with a generic signature like that when it's not preceded by a keyword